traitlets/config/loader.py

```diff
@@ -447,6 +448,52 @@ def __exit__(self, exc_type, exc_value, traceback):
             f.write(json_config)
 
 
+class YAMLFileConfigLoader(FileConfigLoader):
+    """A YAML file loader for config                                                
+    Can also act as a context manager that rewrite the configuration file to disk on exit.                                                                            
+    Example::        
+        with YAMLFileConfigLoader('myapp.yaml','/home/jupyter/configurations/') as c:
+            c.MyNewConfigurable.new_value = 'Updated'     
+    """
+    def load_config(self):
+        """Load the config from a file and return it as a Config object."""
+        self.clear()
+        try:
+            self._find_file()
+        except IOError as e:
+            raise ConfigFileNotFound(str(e))
+        dct = self._read_file_as_dict()
+        self.config = self._convert_to_config(dct)
+        return self.config
+
+    def _read_file_as_dict(self):
+        with open(self.full_filename) as f:
+            return yaml.load(f)
```

Comment:
You most likely want to use `safe_load` instead of `load`, or this may become a security risk.